mantidproject/mantid
# Enable LoadEventNexus to choose load event monitor or histogram monitor

On the SNAP instrument, some of the new files have monitors in both histogram and event mode. However, they don't contain both representations. 
It is found that algorithm LoadNexusMonitor (version 2) cannot read only 1 type of monitors if monitors of both modes exist.  
The work in this PR add two new properties to specify which type (i.e., mode) of monitor to load if both event monitor and histogram monitor are found in NeXus file.

Besides, the codes of algorithm LoadNexusMonitor2 is somehow refactored for better readability.

**To test:**
1. Check whether all the unit tests and system tests are passed.  None of the tests are modified in this work;
2. From system test locate ISIS NeXus file LET00006278.nxs.  Call LoadNexusMonitor to load it with 
   - `MonitorsAsEvents = False`, `LoadEventMonitors = True` and `LoadHistoMonitors=True`, a 8-spectrum workspace should be generated; or
   - set `MonitorsAsEvents = False`, `LoadEventMonitor = False` and `LoadHistoMonitors=True`, the a 6-spectrum workspace should be generated;  or
   - set `MonitorsAsEvents = False`, `LoadEventMonitors = False` and `LoadHistoMonitor=True`, then a 2-spectrum histogram workspace should be generated.
3. Use SNAP_30466_event.nxs for test. There is no way to make this loaded by the old version of LoadNexusMonitor. With the change, you should be able to load it with 
   - `MonitorsAsEvents = False`, `LoadEventMonitors = False` and `LoadHistoMonitors=True`, or 
   - `MonitorsAsEvents = True`, `LoadEventMonitors = True` and `LoadHistoMonitors = False`. 
4. Still use SNAP_30466_event.nxs for test.  Run LoadEventNexus with 
   - `MonitorsAsEvents = False`, `LoadEventMonitors = False` and `LoadHistoMonitors=True`  or
   - `MonitorsAsEvents = True`, `LoadEventMonitors = True` and `LoadHistoMonitors = False`. 



Fixes #15270.

[Release Notes](https://github.com/mantidproject/mantid/blob/15270_load_nexus_monitor_snap/docs/source/release/v3.7.0/framework.rst)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
